app/src/androidTest/java/com/amaze/filemanager/ui/fragments/TabFragmentTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.amaze.filemanager.ui.fragments
+
+import android.content.pm.ActivityInfo
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.swipeLeft
+import androidx.test.espresso.action.ViewActions.swipeRight
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.ActivityTestRule
+import androidx.test.rule.GrantPermissionRule
+import com.amaze.filemanager.R
+import com.amaze.filemanager.ui.activities.MainActivity
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Tests for [TabFragment] functionality, mainly for
+ * https://github.com/TeamAmaze/AmazeFileManager/issues/1555.
+ *
+ * Note: deprecated methods and classes are used here for best reproducing the issues.
+ */
+@Suppress("DEPRECATION")
+@RunWith(AndroidJUnit4::class)
+class TabFragmentTest {
+    @get:Rule
+    val activityRule = ActivityTestRule(MainActivity::class.java)
+
+    @Rule
+    @JvmField
+    val storagePermissionRule: GrantPermissionRule =
+        GrantPermissionRule
+            .grant(android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
+
+    /**
+     * Check if the fragment state is saved correctly during a configuration change
+     * by rotate the screen while swiping between the tabs.
+     */
+    @Test
+    fun testFragmentStateSavingDuringConfigChange() {
+        // First perform the swipe action
+        onView(withId(R.id.pager)).perform(swipeLeft())
+
+        // Force a configuration change by rotating the screen
+        activityRule.activity.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+        Thread.sleep(1000) // Give time for the rotation to complete
```

Comment:
Is using a constant time correct here? Doesn't it depend on the speed of the device whether it crashes or not?